db/corruption_test.cc

```diff
@@ -151,6 +153,7 @@ class CorruptionTest : public testing::Test {
         correct++;
       }
     }
+    iter->status().PermitUncheckedError();
```

Comment:
I thought `PermitUncheckedError()` is for the purpose of temporarily letting some of the tests pass. If I put it in destructor of `CompactionIterator`, then it seems to mean callers of `CompactionIterator` does not have to check its `status_`. By letting caller determine whether and where to check `status_`, we can better enforce the checking. Let me know if I have misunderstood the intention.

Reply:
I think there is generally an issue with Status as member variables.  If I create a CompactionIterator and never do anything with it, I will get an error about the status not being checked.  This seems wrong to me as I never did anything with the iterator. 

In any case, my guess is someone else is going to hit another case where the CompactionIterator status is not checked as well...

Reply:
In the case you mentioned, I think it is easy to just call `PermitUncheckedError()` for the member `status`. This seems a little more, but not the worst thing. It will be good if this can be simplified.

Reply:
Things can be more tedious if the class has multiple status members...